sdk/storage/azure-storage-files-datalake/src/rest_client.cpp

```diff
@@ -114,6 +114,18 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
           {
             vectorElement2.EncryptionScope = var0["EncryptionScope"].get<std::string>();
           }
+          if (var0.count("creationTime") != 0)
+          {
+            vectorElement2.CreatedOn
+                = Core::_internal::Win32FileTimeConverter::Win32FileTimeToDateTime(
+                    std::stoll(var0["creationTime"].get<std::string>()));
+          }
+          if (var0.count("expiryTime") != 0)
+          {
+            vectorElement2.ExpiresOn
+                = Core::_internal::Win32FileTimeConverter::Win32FileTimeToDateTime(
```

Comment:
Expiry time is also in Win32 FILETIME format? Can you confirm?

Reply:
Yes, and I has already added an unit test for this scenario(in DataLakeFileSystemClientTest#ListPaths)

Reply:
If it doesn't expire, this field would be `"0"`, which, after being converted to `Azure::DateTime`, would be `1601-01-01`.
It's not intuitive and is subject to changes in datetime implementation.

We need to do something about it.